ui/src/app/pages/app/component.ts

```diff
@@ -113,9 +122,9 @@ export class AppComponent implements OnInit {
       const currentRoute = this.getLeafRoute();
       if (currentRoute.outlet === 'primary') {
         currentRoute.data.subscribe(value => {
-          const routeTitle = value.title ||
-            decodeURIComponent(currentRoute.params.getValue()[value.pathElementForTitle]);
-          this.titleService.setTitle(`${routeTitle} | ${this.baseTitle}`);
+          const routeTitle = value.title || this.titleFromPathElement(currentRoute, value.pathElementForTitle);
+          this.titleService.setTitle(buildPageTitleForEnvironment(routeTitle));
+          console.log(value, `ng route: ${routeTitle}, ${buildPageTitleForEnvironment(routeTitle)}`)
```

Comment:
Did you mean to leave this `console.log` (and the one below) in? If not, the two lines above could be split out into a function.

Reply:
Thanks - no, that wasn't intentional. Removed.